runtime/src/validate_block/storage_functions.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! All storage functions that are replaced by `validate_block` in the Substrate runtime.
+
+use crate::WitnessData;
+use rstd::{slice, ptr, cmp};
+
+pub static mut STORAGE: Option<WitnessData> = None;
```

Comment:
static mut is very frightening. globals should be protected by some kind of generic guard.

Reply:
the fact that it's public is even scarier. unsafe code should be isolated at module boundaries, where invariants are protected.